### PR TITLE
Docs: Adding a doc on how to propose a cherry-pick

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: docs/development/proposing-a-cherry-pick.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Anyone can propose a cherry-pick - you do not need to be the original author.
+
+We typically have multiple release branches "open" at once - ideally one that is
+a stable release branch, one that is a beta, and one that is an alpha.
+Sometimes the alpha branch will be the `master` branch.
+
+Broad guidelines for acceptance of a cherry-pick:
+
+* Once a version is beta or GA, we typically won't cherry-pick new features back
```
Comment (rifelpet) [inline docs/development/proposing-a-cherry-pick.md]: Maybe its worth explicitly calling out what _**is**_ typically cherry-picked to these versions? eg: bug fixes, regressions, etc.

Comment (mikesplain) [reply docs/development/proposing-a-cherry-pick.md]: Agreed with Peter, we've also discussed a few times in office hours what things we make best effort to cherry pick, e.g. new machine-types, new flag support for components like api-server, etc. 

That said, we probably don't want to cover every angle so leaving it loose is probably useful as well 😄 .

Thanks @justinsb

**Comment by justinsb** [comment]
Thanks for the comments @rifelpet and @mikesplain - totally agree.  Filled out the bullets a little bit to describe what we are likely to accept!

**Comment by rifelpet** [comment]
Awesome!
/lgtm

File: docs/development/proposing-a-cherry-pick.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  that the branches can stabilize.
+
+We are currently tracking cherry-picks to the various branches in a [spreadsheet](https://docs.google.com/spreadsheets/d/1zU67srtZUjuu_9UD7a-mBO6Gp5Z9k4EpYKsJw08_U9c/edit#gid=0)
+
+## Process
+
+The kubernetes repo has a [script for cherry-picking](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh) and you can download the raw form using something like:
+
+```
+wget -O /usr/local/bin/cherry_pick_pull.sh https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/cherry_pick_pull.sh
```
Comment (zetaab) [inline docs/development/proposing-a-cherry-pick.md]: cool. I have done everything manually 👍

**Pull request merged**